ddev/src/ddev/cli/env/stop.py

```diff
@@ -18,42 +18,60 @@
 @click.pass_obj
 def stop(app: Application, *, intg_name: str, environment: str, ignore_state: bool):
     """
-    Stop an environment.
+    Stop environments. To stop all the running environments, use `all` as the integration name and the environment.
     """
     from ddev.e2e.agent import get_agent_interface
     from ddev.e2e.config import EnvDataStorage
     from ddev.e2e.constants import DEFAULT_AGENT_TYPE, E2EEnvVars, E2EMetadata
     from ddev.e2e.run import E2EEnvironmentRunner
     from ddev.utils.fs import temp_directory
 
-    integration = app.repo.integrations.get(intg_name)
-    env_data = EnvDataStorage(app.data_dir).get(integration.name, environment)
-    runner = E2EEnvironmentRunner(environment, app.verbosity)
+    env_data_storage = EnvDataStorage(app.data_dir)
 
-    if not env_data.exists():
-        app.abort(f'Environment `{environment}` for integration `{integration.name}` is not running')
-    elif ignore_state:
-        return
+    if intg_name == 'all':
+        integrations = env_data_storage.get_integrations()
+    else:
+        integrations = [intg_name]
 
-    app.display_header(f'Stopping: {environment}')
+    for integration_name in integrations:
+        if environment == 'all':
+            environments = [
+                env
+                for env in env_data_storage.get_environments(integration_name)
+                if env_data_storage.get(integration_name, env).exists()
+            ]
+        else:
+            environments = [environment]
 
-    # TODO: remove this required result file indicator once the E2E migration is complete
-    with temp_directory() as temp_dir:
-        result_file = temp_dir / 'result.json'
-        env_vars = {E2EEnvVars.RESULT_FILE: str(result_file)}
+        for env in environments:
+            integration = app.repo.integrations.get(integration_name)
```

Comment:
is this `.get` a `dict.get`? Can `integration` be `None`?

Reply:
[Nope](https://github.com/DataDog/integrations-core/blob/0ec46588047cf8d24f2bc4c2096568e22a2c8a82/ddev/src/ddev/repo/core.py#L78-L91)

Reply:
Nice! not a great method choice but that's another topic :)